.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,6 +1,9 @@
-Fixes #<issue_number_goes_here>
+There's no need to add anything here if you use a single commit per logical
```

Comment:
If you enclose this with an HTML comment, then the writer doesn't have to delete it, as it won't show up in the PR rendering. Like this:

```
<!--
Instructions here
-->
```

That being said, I think it's fair for any given project to enforce its preferred workflow on contributors, so you could just pick one and mandate it.

Reply:
> If you enclose this with an HTML comment, then the writer doesn't have to delete it, as it won't show up in the PR rendering. Like this:
> 
> ```
> <!--
> Instructions here
> -->
> ```

Good idea, done.

> That being said, I think it's fair for any given project to enforce its preferred workflow on contributors, so you could just pick one and mandate it.

Sounds good. I've updated it to say that we prefer the commit-per-logical-change workflow.

Reply:
Now that I've used GitHub's PRs for a few days, I've realized how poor the support for force-pushing is :( What bothers me most is that GitHub simply forgets commits pointed to by previous versions of a PR.

I'm considering using https://reviewable.io/ for this project. I haven't tried it yet, though. Reviewable adds refs under `refs/reviewable/` to prevent GC. It also (supposedly) has better understanding of rebasing in general, such as by not showing (or dimming?) parts of the diff that are a result of rebasing.

Reply:
I'll merge this PR now even though I'm not sure how to deal with GitHub's poor support for force-pushes.